src/Microsoft.PowerShell.Commands.Utility/commands/utility/MarkdownOptionCommands.cs

```diff
@@ -173,11 +175,23 @@ protected override void EndProcessing()
                     break;
             }
 
-            this.CommandInfo.Module.SessionState.PSVariable.Set(MarkdownOptionInfoVariableName, mdOptionInfo);
+            PSMarkdownOptionInfo setOption = null;
+
+            // If we have the moduleInfo then store are module scope variable
+            if (this.CommandInfo.Module != null)
+            {
+                this.CommandInfo.Module.SessionState.PSVariable.Set(MarkdownOptionInfoVariableName, mdOptionInfo);
+                setOption = mdOptionInfo;
+            }
+            else // If we don't have a moduleInfo, like in PowerShell hosting scenarios, use a concurrent dictionary.
```

Comment:
I guess CodeFactor will reports a style issue for the comment.
We could move it before line 188.

Reply:
As mentioned above, comments need to be make more clear.

Reply:
Refactored this.